### ISSUE TITLE
Can't activate Site editor edit mode by using the keyboard from the preview iframe any longer

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#57400]
### Description

This appears to be a regression in WordPress 6.4.. It works as expected in WordPress 6.3.1 (gutenberg 15.2 - 16.1).

### Step-by-step reproduction instructions

WordPress 6.3.1 
- Go to the Site editor, make sure the initial view is the 'view mode', the one that shows the navigation panel.
- Use the keyboard to navigate the user interface
- When focus is on the preview iframe, a blue outline appears around it.
- Press the Enter or Spacebar keys.
- Observe the editor switches to 'edit mode'.

WordPress 6.4 or latest trunk
- Repeat the steps above.
- Observe the editor does _not_ switch to edit mode.
- Observe that when usign the Spacebar key, the iframe content scrolls down instead.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by afercia** [comment]
I haven't been able to track down what the root cause of this regression is.
 Right now, the situation is less than ideal we we're left with an iframe that:
 - It's focusable because it has a `tabindex="0"` so it's a tab stop that doesn't do anything.
 - The iframe semantics isi overridden by a `role="button"`

 It's basically a hack we made to provide keyboard users a way to switch the editor to edit mode. Arguably, we should not use such a hack and we should not override the iframe semantics. Instead of fixing it, I'd suggest to _change_ the mechanism and provide a button that is only revealed on focus.

Note:
this keydown event on the iframe appears to not be executed at all, while the click event is:

https://github.com/WordPress/gutenberg/blob/1129dcf586535923708055adc67c924685bcad5c/packages/edit-site/src/components/block-editor/editor-canvas.js#L55-L70

**Comment by stokesman** [comment]
> Instead of fixing it, I'd suggest to _change_ the mechanism and provide a button that is only revealed on focus.

To me a dedicated button sounds good. Perhaps this issue should be revised to focus on that suggestion. The issue as stated in the title seems to be fixed.

Also, it sounds like a dedicated button might be the most straightforward fix for #59319

**Comment by afercia** [comment]
> The issue as stated in the title seems to be fixed.

Yes it seems fixed on latest trunk, thanks for testing it. I'll work on adding an e2e test to make sure it doesn't regress again.

No objections with repurposing the scope of this issue to add a dedicated button.

**Comment by afercia** [comment]
> I'll work on adding an e2e test to make sure it doesn't regress again.

The expected behavior is already tested here:

https://github.com/WordPress/gutenberg/blob/acb176ef83c704e36144a59303d59551a00f2145/test/e2e/specs/site-editor/navigation.spec.js#L76-L84

I'm not sure I like where initla initial lands though and the test could be a little more descriptive.

**Comment by afercia** [comment]
The `<iframe>` with a `role="button"` is fundamentally a big hack that should be avoided in the first place. Is there any historical reason why we're not using a more standard button, for example a transparent button overlaid on top of the entire iframe preview?

As mentioned on https://github.com/WordPress/gutenberg/issues/57400#issuecomment-2704911716 a dedicated button might also be the most straightforward fix for #59319.
